### PR TITLE
Hv unit test fetcher

### PR DESCRIPTION
**Pull request by HansVRP**
some unit test for the s2 fetchers and the underlying functions in common

### CODE REVIEW

**Review by VincentVerelst — COMMENTED** [review]
As a general comment: should we keep the same folder structure in `tests` as in `src`?

File: tests/test_openeo_gfmap/test_commons.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    convert_band_names,
+    rename_bands,
+    resample_reproject,
+)
+from openeo_gfmap.fetching.s2 import (
+    BASE_SENTINEL2_L2A_MAPPING,
+    ELEMENT84_SENTINEL2_L2A_MAPPING,
+)
+
+from .utils import create_test_datacube
```
[the remark this answers is not in the document]
Comment (HansVRP) [reply tests/test_openeo_gfmap/test_commons.py:16]: I agree that that's a better place to define this function

File: src/openeo_gfmap/fetching/s2.py [hunk 2]
```diff
@@ -113,6 +113,7 @@ def s2_l2a_fetch_default(
     return s2_l2a_fetch_default
 
 
+# TODO deprecated?
```
[the remark this answers is not in the document]
Comment (HansVRP) [reply src/openeo_gfmap/fetching/s2.py:116]: i'll add a deprecated indicator

File: tests/test_openeo_gfmap/test_unit_collection_fetcher.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    """Test that CollectionFetcher raises an error with empty bands."""
+    bands = []
+    fetcher = CollectionFetcher(
+        backend_context=mock_backend_context,
+        bands=bands,
+        collection_fetch=MagicMock(),  # No need to mock fetch here
+        collection_preprocessing=MagicMock(),
+    )
+
+    # with pytest.raises(ValueError, match="Bands cannot be empty"):
```
[the remark this answers is not in the document]
Comment (HansVRP) [reply tests/test_openeo_gfmap/test_unit_collection_fetcher.py:141]: let me check if this was some work in progress code that snuck in

**Comment by HansVRP** [comment]
> As a general comment: should we keep the same folder structure in `tests` as in `src`?

@VincentVerelst  I think this is a good suggestion. 

@GriffinBabe will this screw over the current jenkins pipeline? (as in, does it look in 'tests' or 'test_openeo_gfmap')

**Comment by HansVRP** [comment]
I suggest that we rearrange the test in a follow-up PR. @VincentVerelst let me know if you can approve

**Review by VincentVerelst — APPROVED** [review]

**Pull request merged**